### PR TITLE
Migrating scope stages into handlers

### LINKED ISSUE #1054: Migrate `document` to use a scope handler

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
Fixes #1054

## Checklist

- [ ] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [ ] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet_html)

### CODE REVIEW

File: src/processTargets/modifiers/scopeHandlers/LineScopeHandler.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ export default class LineScopeHandler implements ScopeHandler {
   public readonly iterationScopeType = { type: "document" } as const;
 
   constructor(
-    public readonly scopeType: { type: "line" },
+    public readonly scopeType: ScopeType,
```
Comment (pokey) [inline src/processTargets/modifiers/scopeHandlers/LineScopeHandler.ts:15]: I'm thinking of using this type information for generating our docs.  Do you mind if we leave it?

Comment (AndreasArvidsson) [reply src/processTargets/modifiers/scopeHandlers/LineScopeHandler.ts:15]: Why not turn it into a proper type instead? I mean in the types file we have made them all `SimpleScopeType` if we actually want to have it more fine grain I think we should do it at that end so we don't have to do casts when calling the constructor.
`public readonly scopeType: LineScopeType`

Comment (pokey) [reply src/processTargets/modifiers/scopeHandlers/LineScopeHandler.ts:15]: Why would giving it a name allow us to avoid casts?

**Comment by pokey** [comment]
Folded into #1075

**Pull request closed without merging**